src/python/module/smartredis/client.py

```diff
@@ -43,50 +42,60 @@
+        order was swapped for first two parameters in the second signature
+        relative to previous releases of SmartRedis; this was necessary to
+        remove ambiguity.
 
             Client(config_options: ConfigOptions=None,
                    logger_name: str="Default")
             Client(cluster: bool, address: optional(str)=None,
-                   logger_name: str="Default") <= Deprecated!
+                   logger_name: str="Default")
 
         For detailed information on the first signature, please refer
-        to the __new_construction() method below.
+        to the __standard_construction() method below.
 
         For detailed information on the second signature, please refer
-        to the __deprecated_construction() method below.
+        to the __address_construction() method below.
 
-        :param a: The positional arguments supplied to this method; see above for
-                  valid options
+        :param a: The positional arguments supplied to this method;
+                  see above for valid options
         :type a: tuple[any]; see above for valid options
-        :param kw: Keyword arguments supplied to this method; see above for
-                   valid options
+        :param kw: Keyword arguments supplied to this method;
+                   see above for valid options
         :type kw: dict[string, any]; see above for valid options
         :raises RedisConnectionError: if connection initialization fails
         """
         if a:
             if isinstance(a[0], bool):
-                pyclient = self.__deprecated_construction(*a, **kw)
+                for arg in kw:
+                    if arg not in ["cluster", "address", "logger_name"]:
+                        raise TypeError(
+                            f"__init__() got an unexpected keyword argument '{arg}'"
+                        )
+                pyclient = self.__address_construction(*a, **kw)
             elif isinstance(a[0], ConfigOptions) or a[0] is None:
-                pyclient = self.__new_construction(*a, **kw)
+                pyclient = self.__standard_construction(*a, **kw)
             else:
                 raise TypeError(f"Invalid type for argument 0: {type(a[0])}")
         else:
-            config_object = kw.get("config_object", None)
-            logger_name = kw.get("logger_name", "default")
-            pyclient = self.__new_construction(config_object, logger_name)
+            # Only kwargs in the call
+            if "address" in kw:
+                pyclient = self.__address_construction(*a, **kw)
+            else:
+                pyclient = self.__standard_construction(*a, **kw)
         super().__init__(pyclient)
 
-    def __deprecated_construction(self, cluster, address=None, logger_name="Default"):
-        """Initialize a RedisAI client (Deprecated)
+    def __address_construction(self, cluster, address=None, logger_name="Default"):
```

Comment:
Not desperately important, but if we could retroactively type hint this method that might be nice for future developers
```py
    def __address_construction(
        self,
        cluster: bool,
        address: t.Optional[str] = None,
        logger_name: str = "Default"
    ) -> PyClient:
```

Reply:
Done